moveit_planners/stomp/src/stomp_moveit_smoothing_adapter.cpp

```diff
@@ -0,0 +1,125 @@
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of PickNik Inc. nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Henning Kayser */
+
+// ROS
+#include <rclcpp/rclcpp.hpp>
+#include <class_loader/class_loader.hpp>
+
+// MoveIt
+#include <moveit/planning_request_adapter/planning_request_adapter.h>
+#include <moveit/planning_interface/planning_interface.h>
+
+// STOMP
+#include <stomp_moveit/stomp_moveit_planning_context.hpp>
+
+using namespace planning_interface;
+
+namespace stomp_moveit
+{
+class StompSmoothingAdapter : public planning_request_adapter::PlanningRequestAdapter
+{
+public:
+  StompSmoothingAdapter() : planning_request_adapter::PlanningRequestAdapter()
+  {
+  }
+
+  // todo[noetic] add override again
+  void initialize(const rclcpp::Node::SharedPtr& node, const std::string& parameter_namespace) override
+  {
+    param_listener_ = std::make_shared<stomp_moveit::ParamListener>(node, parameter_namespace);
+  }
+
+  std::string getDescription() const override
+  {
+    return "Stomp Smoothing Adapter";
+  }
+
+  bool adaptAndPlan(const PlannerFn& planner, const planning_scene::PlanningSceneConstPtr& ps,
+                    const planning_interface::MotionPlanRequest& req, planning_interface::MotionPlanResponse& res,
+                    std::vector<std::size_t>& /*added_path_index*/) const override
+  {
+    // Following call to planner() calls the OMPL planner and stores the trajectory inside the MotionPlanResponse res
```

Comment:
Why and which OMPL planner? Shouldn't the smoothing adapter be agnostic to the planner it is used with?